peps/pep-0735.rst

```diff
@@ -202,12 +201,15 @@ Specification
 This PEP defines a new section (table) in ``pyproject.toml`` files named
 ``dependency-groups``. The ``dependency-groups`` table contains an arbitrary
 number of user-defined keys, each of which has, as its value, a list of
-requirements (defined below).  These keys must match the following
-regular expression: ``[a-z0-9][a-z0-9-]*[a-z0-9]``. Meaning that they must be
-all lower-case alphanumerics, with ``-`` allowed only in the middle, and at
-least two characters long. These requirements are chosen so that the
-normalization rules used for PyPI package names are unnecessary as the names
-are already normalized.
+requirements (defined below). These keys must be
+`valid non-normalized names <https://packaging.python.org/en/latest/specifications/name-normalization/#valid-non-normalized-names>`__,
+and must be
+`normalized <https://packaging.python.org/en/latest/specifications/name-normalization/#normalization>`__
+before comparisons.
+
+Tools SHOULD prefer to present the original, non-normalized name to users by
```

Comment:
This seems like a UI choice that needn't be specified by the PEP.

Reply:
Not sure if you raised this suggestion/nitpick elsewhere?
FWIW, for a tool that converts the metadata to an alternative format, this is a useful hint that it might want to preserve the non-normalized name.

Reply:
This was the suggested text in the discuss thread and I took it more or less verbatim.

Let's take this into the discuss thread if we think it's still an issue. FWIW, I don't think the reference implementation fully follows this recommendation.